core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -511,17 +511,19 @@ private final QueryAssert hasPlan(PlanMatchPattern expectedPlan, Consumer<Plan>
 
             if (!skipResultsCorrectnessCheckForPushdown) {
                 // Compare the results with pushdown disabled, so that explicit matches() call is not needed
-                verifyResultsWithPushdownDisabled();
+                hasCorrectResultsRegardlessOfPushdown();
             }
             return this;
         }
 
-        private void verifyResultsWithPushdownDisabled()
+        @CanIgnoreReturnValue
+        public QueryAssert hasCorrectResultsRegardlessOfPushdown()
```

Comment:
Imho this method explicitly tests result without a pushdown so name is kind of missleading

Reply:
it compares "results" (weaved into `this`) with "results without pushdown" (obtained within this method)